index.ts

```diff
@@ -0,0 +1,72 @@
+import { spawn, SpawnOptionsWithoutStdio } from "child_process";
+import type { Plugin as VitePlugin } from "vite";
+
+// Utility to invoke a given sbt task and fetch its output
+function printSbtTask(task: string, cwd?: string): Promise<string> {
+  const args = ["--batch", "-no-colors", "-Dsbt.supershell=false", `print ${task}`];
+  const options: SpawnOptionsWithoutStdio = {
+    cwd: cwd,
+  };
+  const child = process.platform === 'win32'
+    ? spawn("sbt.bat", args.map(x => `"${x}"`), {shell: true, ...options})
+    : spawn("sbt", args, options);
+
+  let fullOutput: string = '';
+
+  child.stdout.setEncoding('utf-8');
+  child.stdout.on('data', data => {
+    fullOutput += data;
+    process.stdout.write(data); // tee on my own stdout
+  });
+
+  child.stderr.setEncoding('utf-8');
+  child.stderr.on('data', data => {
+    process.stderr.write(data); // tee on my own stderr
+  });
+
+  return new Promise((resolve, reject) => {
+    child.on('close', code => {
+      if (code !== 0)
+        reject(new Error(`sbt invocation for Scala.js compilation failed with exit code ${code}`));
+      else
+        resolve(fullOutput.trimEnd().split('\n').at(-1)!);
+    })
+  })
+}
+
+export interface ScalaJSPluginOptions {
+  projectID?: string,
+  cwd?: string,
+}
+
+export default function scalaJSPlugin(options: ScalaJSPluginOptions = {}): VitePlugin {
+  const { projectID, cwd } = options;
+
+  let isDev = false; // when resolving the Vite config, this may turn into true
```

Comment:
Consider being stricter here and type `isDef: boolean | undefined`. Then fail if `configResolved` didn't get called.